src/riscv/lib/src/jit/builder/ext_calls.rs

```diff
@@ -0,0 +1,139 @@
+// SPDX-FileCopyrightText: 2025 TriliTech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+//! Framework for calling external functions from Cranelift IR
+
+use cranelift::codegen::ir;
+use cranelift::prelude::FunctionBuilder;
+use cranelift::prelude::InstBuilder;
+use cranelift::prelude::Signature;
+use cranelift::prelude::isa::TargetFrontendConfig;
+
+use crate::jit::builder::typed::Type;
+use crate::jit::builder::typed::Typed;
+use crate::jit::builder::typed::Value;
+
+/// Trait for types that can be returned from external function calls
+pub trait ReturnTyped {
+    /// IR types of the function return values
+    const RETURN_TYPES: &[Type];
```

Comment:
I don't quite get how `RETURN_TYPES` works. In what cases would this be >1 ? What does it mean for a function to have multiple return types?

Reply:
There are a few cases where you can have multiple return values. Ignoring that Cranelift itself allows you to return multiple values from a function, you can have multiple return values from external functions per the C ABI. Struct(ured) returns can be used to return up to two values, I think. Generally, Cranelift doesn't constrain the number of return values, though.